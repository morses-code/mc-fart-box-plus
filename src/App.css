.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.app-header-container {
  text-align: center;
  padding: 1rem;
  background-color: #fff;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.1);
  margin-bottom: 1rem;
}

.app-header {
  font-size: 3rem;
  margin-bottom: 1rem;
  color: #333;
  text-shadow: 1px 1px 3px rgba(0, 0, 0, 0.2);
  
}

.app-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin-top: -1rem;
  padding-top: 1rem; 
  padding-bottom: 1rem; 
}

.sound-button {
  font-size: 10rem;
  margin: 0.5rem;
  padding: 1rem;
  width: calc(33.33% - 1rem);
  max-width: 250px;
  background-color: #53ac56;
  color: #fff;
  border: none;
  border-radius: 8px;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.3);
}

.sound-button:hover {
  background-color: #45a049;
}

.sound-button:active {
  transform: translateY(2px);
  box-shadow: none;
}


@media (max-width: 600px) {
  .app-header {
    font-size: 2rem;
    margin-bottom: 0.5rem; /* Adjust the bottom margin as needed */
  }

  .sound-button {
    width: calc(50% - 1rem);
  }
}

