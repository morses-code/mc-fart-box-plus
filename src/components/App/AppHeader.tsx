/**
 * @file AppHeader.js
 * @description Contains the AppHeader component for the header section of the app.
 */

import React from 'react';

/**
 * @component AppHeader
 * @description A component that represents the header section of the app.
 * @return {JSX.Element} The rendered AppHeader component.
 */
const AppHeader: React.FC = () => {
	return (
		<div>
			<img
				src='/logo-no-background.png'
				alt='Logo'
				className='header-logo'
			/>
		</div>
	);
};

export default AppHeader;
