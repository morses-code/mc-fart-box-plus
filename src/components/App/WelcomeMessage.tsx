const WelcomeMessage = () => (
	<div className='welcome'>
		<p>Welcome to Fart Box Plus!</p>
		<p>
			Experience the ultimate collection of funny fart sounds in the form of a
			virtual sound box. Our app brings you hours of laughter and entertainment
			with a wide variety of hilarious fart noises. From classic to outrageous,
			we have them all!
		</p>
		<p>
			But that's not all - Fart Box Plus is not just about sounds. We're
			constantly expanding our offerings to bring you even more fart-related
			fun! In the near future, we have exciting plans to introduce games like
			Pong and Snake, all with a unique and amusing fart-related twist. Get
			ready for some seriously silly gaming experiences!
		</p>
		<p>
			Whether you're looking to prank your friends, add some humor to your day,
			or simply indulge in the joy of childish laughter, Fart Box Plus has got
			you covered. We guarantee endless amusement and a good time for everyone.
		</p>
		<p>
			So, buckle up and embark on this gas-powered adventure with Fart Box Plus.
			Stay tuned for updates and new features as we continue to expand our
			fart-filled world of laughter and entertainment!
		</p>
	</div>
);

export default WelcomeMessage;
