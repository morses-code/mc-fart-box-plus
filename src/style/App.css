:root {
	--primary-color: #096a2e;
	--secondary-color: #8abe53;
	--text-color: #e9c893;
	--background-color: #ffffff;
}

.app-header-container {
	width: 100%;
	max-width: 900px;
	margin: 0 auto;
	padding: 20px;
	box-sizing: border-box;
	justify-content: center;
}

.app-container {
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
	margin: 0 auto;
	padding: 20px;
}

.welcome {
	background-color: var(--primary-color);
	border-radius: 10px;
}

.header {
	font-size: 32px;
	font-weight: bold;
	letter-spacing: 2px;
	line-height: 1.2;
	color: var(--text-color);
	text-align: center;
}

.header-logo {
	display: block;
	margin: 0 auto;
	max-width: 100%;
	height: auto;
	padding: 10px 0 10px 0;
	text-align: center;
}

.navbar-list {
	list-style-type: none;
	margin: 0;
	padding: 0;
	display: flex;
	justify-content: center;
	align-items: center;
	background-color: var(--primary-color);
	border-radius: 10px;
}

.navbar-list li {
	margin: 0 10px;
}

.navbar-list li a {
	font-size: 18px;
	display: block;
	padding: 10px 20px;
	text-decoration: none;
	border-radius: 10px;
	transition: background-color 0.3s ease;
	text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5);
	text-align: center;
}

.navbar-list li a:hover {
	background-color: var(--secondary-color);
}

.navbar-list li a:visited {
	color: var(--text-color);
}

.sound-button {
	display: flex;
	align-items: center;
	justify-content: center;
	width: 100%;
	height: 100px;
	font-size: 32px;
	background-color: var(--primary-color);
	border: none;
	border-radius: 8px;
	margin: 10px;
	cursor: pointer;
	transition: background-color 0.3s ease;
	max-width: 100px;
	width: 33.33%;
}

.sound-button:hover {
	background-color: var(--secondary-color);
}

p {
	padding: 5px;
	margin: 0.5rem 0;
	font-size: 1rem;
	line-height: 1;
	color: var(--text-color);
	word-wrap: break-word; /* Enable word wrapping */
	overflow-wrap: break-word; /* Alternative property for word wrapping */
}

@media (max-width: 768px) {
	.sound-button {
		flex-basis: calc(50% - 20px);
		width: 50%;
	}

	p {
		font-size: 0.9rem;
		word-wrap: break-word;
	}
}

@media (max-width: 480px) {
	.sound-button {
		flex-basis: calc(100% - 20px);
		width: 50%;
	}

	p {
		font-size: 0.8rem;
	}
}

@font-face {
	font-family: 'Advent Pro Medium';
	src: url('/public/fonts/Advent-Pro-Medium.ttf');
}
