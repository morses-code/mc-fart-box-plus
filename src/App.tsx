import React, { useState } from 'react';
import './App.css';
import SoundButton from './SoundButton';
import AppHeader from './AppHeader';
import NavigationBar from './NavigationBar';

const App: React.FC = () => {
	const soundSets = [
		[
			['ðŸ’©'],
			[
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound1.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound2.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound3.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound4.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound5.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound6.mp3',
			],
		],
		[
			['ðŸ˜¯'],
			[
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion1.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion2.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion3.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion4.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion5.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion6.mp3',
			],
		],
	];
	const [currentSet, setCurrentSet] = useState(0);

	const handleSetChange = (setIndex: number) => {
		setCurrentSet(setIndex);
	};

	return (
		<div>
			<div className='app-header-container'>
				<AppHeader />
			</div>
			<NavigationBar
				onSetChange={handleSetChange}
				soundSets={soundSets}
				currentSet={currentSet}
			/>
			<div className='app-container'>
				{soundSets[currentSet]
					.filter((_, index) => index !== 0)
					.map((soundFile, index) =>
						soundFile.map((sounds) => (
							<SoundButton
								key={index}
								soundFile={sounds}
								text={soundSets[currentSet][0][0]}
							/>
						))
					)}
			</div>
		</div>
	);
};

export default App;
