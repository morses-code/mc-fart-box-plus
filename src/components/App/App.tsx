/**
 * @file App.js
 * @description Contains the main App component and its dependencies.
 */

import React, { useState } from 'react';
import '../../style/App.css';
import SoundButton from '../SoundButton/SoundButton';
import AppHeader from './AppHeader';
import NavigationBar from '../NavigationBar/NavigationBar';

/**
 * @component App
 * @description The main component of the application.
 * @returns {JSX.Element} The rendered App component.
 */
const App: React.FC = () => {
	const soundSets = [
		[
			['ðŸ’©', 'Fart Sounds'],
			[
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound1.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound2.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound3.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound4.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound5.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/sound6.mp3',
			],
		],
		[
			['ðŸ˜¯', 'Minions Sounds'],
			[
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion1.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion2.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion3.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion4.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion5.mp3',
				'https://storage.googleapis.com/fart-box-387618.appspot.com/minion6.mp3',
			],
		],
	];

	const [currentSet, setCurrentSet] = useState(0);
	const [audio, setAudio] = useState<HTMLAudioElement | null>(null);

	/**
	 * @function handleSetChange
	 * @description Updates the current set index when a set item is clicked.
	 * @param {number} setIndex - The index of the clicked set item.
	 * @returns {void}
	 */
	const handleSetChange = (setIndex: number) => {
		setCurrentSet(setIndex);
	};

	return (
		<div>
			<div className='app-header-container'>
				<AppHeader />
				<div className='welcome'>
					<p>Welcome to Fart Box Plus!</p>
					<p>
						This is a place for gross fart related fun, suitable for everyone.
					</p>
					<p>
						Please enjoy and get in touch with any ideas you would like to see.
					</p>
					<p>Use the navigation buttons below to select sound groups</p>
					<p>This is a virtual sound box for all things gross!</p>
				</div>
				<NavigationBar
					onSetChange={handleSetChange}
					soundSets={soundSets}
					currentSet={currentSet}
				/>
				<div className='app-container'>
					{soundSets[currentSet]
						.filter((_, index) => index !== 0)
						.map((soundFile, setIndex) =>
							soundFile.map((sounds, buttonIndex) => (
								<SoundButton
									key={`${setIndex}-${buttonIndex}`}
									soundFile={sounds}
									text={soundSets[currentSet][0][0]}
									audio={audio}
									setAudio={setAudio}
								/>
							))
						)}
				</div>
				<div className='welcome'>
					<p>Coming soon!</p>
					<p>Fart related games: </p>
					<p>Pong</p>
					<p>Poo Snake</p>
				</div>
			</div>
		</div>
	);
};

export default App;
