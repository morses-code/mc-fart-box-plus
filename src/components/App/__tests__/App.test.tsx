import React from 'react';
import { render, fireEvent, getByAltText } from '@testing-library/react';
import App from '../App';

describe('App', () => {
	const mockPlay = jest.fn();
	const mockPause = jest.fn();

	beforeEach(() => {
		jest.spyOn(HTMLMediaElement.prototype, 'play').mockImplementation(mockPlay);
		jest
			.spyOn(HTMLMediaElement.prototype, 'pause')
			.mockImplementation(mockPause);
	});

	afterEach(() => {
		mockPlay.mockRestore();
		mockPause.mockRestore();
	});

	it('renders the app', () => {
		const { container } = render(<App />);

		expect(container).toMatchSnapshot();
	});

	it('renders the app header', () => {
		const { getByAltText } = render(<App />);
		const headerElement = getByAltText('Logo');

		expect(headerElement).toBeInTheDocument();
	});

	it('renders sound buttons with correct props', () => {
		const { container } = render(<App />);
		const soundButtons = container.querySelectorAll('.sound-button');

		expect(soundButtons.length).toBe(6);

		soundButtons.forEach((button, index) => {
			const soundFile = `https://storage.googleapis.com/fart-box-387618.appspot.com/sound${
				index + 1
			}.mp3`;
			expect(button).toHaveAttribute('data-sound-file', soundFile);
		});
	});

	it('handleSetChange updates currentSet correctly', () => {
		const { getByTestId } = render(<App />);
		const set1Button = getByTestId('set1-button');
		const set2Button = getByTestId('set2-button');

		fireEvent.click(set1Button);
		expect(set1Button).toHaveClass('active');
		expect(set2Button).not.toHaveClass('active');

		fireEvent.click(set2Button);
		expect(set1Button).not.toHaveClass('active');
		expect(set2Button).toHaveClass('active');
	});

	it('plays audio when sound button is clicked', () => {
		const { container } = render(<App />);
		const soundButtons = container.querySelectorAll('.sound-button');

		const audioPlaySpy = jest.spyOn(window.HTMLMediaElement.prototype, 'play');
		const audioPauseSpy = jest.spyOn(
			window.HTMLMediaElement.prototype,
			'pause'
		);

		fireEvent.click(soundButtons[0]);

		expect(audioPlaySpy).toHaveBeenCalledTimes(1);

		fireEvent.click(soundButtons[0]);

		expect(audioPauseSpy).toHaveBeenCalledTimes(1);

		audioPlaySpy.mockRestore();
		audioPauseSpy.mockRestore();
	});
});
