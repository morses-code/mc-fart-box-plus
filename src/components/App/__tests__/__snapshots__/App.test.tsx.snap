// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app 1`] = `
<div>
  <div>
    <div
      class="app-header-container"
    >
      <div>
        <img
          alt="Logo"
          class="header-logo"
          src="/logo-no-background.png"
        />
      </div>
      <nav
        class="navbar"
      >
        <ul
          class="navbar-list"
        >
          <li>
            <a
              class=""
              data-testid="set1-button"
              href="#set0"
            >
              Fart Sounds
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="welcome"
      >
        <p>
          Welcome to Fart Box Plus!
        </p>
        <p>
          Experience the ultimate collection of funny fart sounds in the form of a virtual sound box. Our app brings you hours of laughter and entertainment with a wide variety of hilarious fart noises. From classic to outrageous, we have them all!
        </p>
        <p>
          But that's not all - Fart Box Plus is not just about sounds. We're constantly expanding our offerings to bring you even more fart-related fun! In the near future, we have exciting plans to introduce games like Pong and Snake, all with a unique and amusing fart-related twist. Get ready for some seriously silly gaming experiences!
        </p>
        <p>
          Whether you're looking to prank your friends, add some humor to your day, or simply indulge in the joy of childish laughter, Fart Box Plus has got you covered. We guarantee endless amusement and a good time for everyone.
        </p>
        <p>
          So, buckle up and embark on this gas-powered adventure with Fart Box Plus. Stay tuned for updates and new features as we continue to expand our fart-filled world of laughter and entertainment!
        </p>
      </div>
      <div
        class="welcome"
      >
        <p>
          Coming soon!
        </p>
        <p>
          Fart related games: 
        </p>
        <p>
          Pong
        </p>
        <p>
          Poo Snake
        </p>
      </div>
    </div>
  </div>
</div>
`;
